src/test/frameworks/compose/discover/runtime/node.spec.ts

```diff
@@ -0,0 +1,322 @@
+          integrity sha512-abc123...
+        "@algolia/autocomplete-core@npm:1.7.1":
+          version "npm:1.7.1"
+          resolved "https://registry.yarnpkg.com/ms/-/ms-2.1.3.tgz"
+          integrity sha512-xyz456...
+        "mocha@^9.1.1":
+          version "9.1.1"
+          resolved "https://registry.yarnpkg.com/typescript/-/typescript-4.0.2.tgz"
+          integrity sha512-pqr789...
+      `,
+      });
+      const packageJSON: PackageJSON = {
+        dependencies: {
+          express: "^4.18.2",
+        },
+        devDependencies: {
+          "@algolia/autocomplete-core": "npm:1.7.1",
+          mocha: "^9.1.1",
+        },
+      };
+      const actual = await nodeJSRuntime.getDependencies(fileSystem, packageJSON, "yarn");
+      const expected = {
+        express: "^4.18.2",
+        "@algolia/autocomplete-core": "npm:1.7.1",
+        mocha: "^9.1.1",
+      };
+
+      expect(actual).to.deep.equal(expected);
+    });
+  });
+
+  describe("detectedCommands", () => {
+    it("scripts are run using yarn and framework commands prepend npx", () => {
+      const matchedFramework: FrameworkSpec = {
+        id: "next",
+        runtime: "nodejs",
+        requiredDependencies: [],
+        commands: {
+          dev: {
+            cmd: "next dev",
+            env: { NODE_ENV: "dev" },
+          },
+        },
+      };
+      const scripts = {
+        build: "next build",
+        start: "next start",
+      };
+
+      const actual = nodeJSRuntime.detectedCommands("yarn", scripts, matchedFramework);
+      const expected = {
+        build: {
+          cmd: "yarn run build",
+        },
+        dev: {
+          cmd: "npx next dev",
+          env: { NODE_ENV: "dev" },
+        },
+        run: {
+          cmd: "yarn run start",
+          env: { NODE_ENV: "production" },
+        },
+      };
+
+      expect(actual).to.deep.equal(expected);
+    });
+
+    it("scripts have higher preference over framework commands", () => {
+      const matchedFramework: FrameworkSpec = {
+        id: "next",
+        runtime: "nodejs",
+        requiredDependencies: [],
+        commands: {
+          build: {
+            cmd: "next build testing",
+          },
+          run: {
+            cmd: "next start testing",
+            env: { NODE_ENV: "production" },
+          },
+          dev: {
+            cmd: "next dev",
+            env: { NODE_ENV: "dev" },
+          },
+        },
+      };
+      const scripts = {
+        build: "next build",
+        start: "next start",
+      };
+
+      const actual = nodeJSRuntime.detectedCommands("yarn", scripts, matchedFramework);
+      const expected = {
+        build: {
+          cmd: "yarn run build",
+        },
+        dev: {
+          cmd: "npx next dev",
+          env: { NODE_ENV: "dev" },
+        },
+        run: {
+          cmd: "yarn run start",
+          env: { NODE_ENV: "production" },
+        },
+      };
+
+      expect(actual).to.deep.equal(expected);
```

Comment:
q: this test looks pretty identical to the previous test case 🤔  what is the difference?

Reply:
the preference is given to `commands present in scripts` over `commands present in framework specs`.